### PR TITLE
Refactor tests, splitting out Consolonia.NUnit library and a ton of unit tests 

### PR DESCRIPTION
**Pull request by tomlm**
* Split unit tests out from consolonia support package
* renamed  test support library to Consolonia.NUnit and turned on package
* renamed Consolonia.GalleryTests to Consolonia.Gallery.Tests
* Created Consolonia.Core.Tests library
* Added unit tests for
  * Pixels
  * PixelBackground
  * PixelForeground
  * SimpleSymbol
  * DrawingBoxSymbol
  * Glyph calculation
  * Colors
* Refactored Gallery.Tests to Move type attribute up into gallery test because the way it was set up you couldn't double on a unit test and go to the unit test, and you could only have 1 unit test per class.
image image

## Summary by CodeRabbit

- **New Features**
	- Enhanced text rendering capabilities, including improved handling of glyphs and complex emojis.
	- New equality comparison methods for several structs, including `DrawingBoxSymbol`, `Pixel`, `PixelBackground`, and `PixelForeground`.
	- Added support for complex emoji in text shaping and glyph retrieval.

- **Bug Fixes**
	- Corrected spelling of `StrikethroughThickness` in the drawing context.

- **Tests**
	- Introduced comprehensive unit tests for various classes, including `DrawingBoxSymbol`, `Glyphs`, `Pixel`, and `PixelForeground`.
	- Refactored existing tests for improved clarity and organization.
	- Enhanced accessibility of test methods across multiple test classes.

- **Chores**
	- Updated project references and assembly attributes for internal visibility adjustments.
	- Organized solution structure and updated namespace references across multiple test files.

### CODE REVIEW

**Comment by tomlm** [comment]
@coderabbitai review

File: src/Tests/Consolonia.Core.Tests/PixelBackgroundTests.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+            Assert.That(pixelBackground.Mode, Is.EqualTo(PixelBackgroundMode.Colored));
+        }
+
+        [Test]
+        public void ConstructorWithMode()
+        {
+            var pixelBackground = new PixelBackground(PixelBackgroundMode.Shaded, Colors.Red);
+            Assert.That(pixelBackground.Color, Is.EqualTo(Colors.Red));
+            Assert.That(pixelBackground.Mode, Is.EqualTo(PixelBackgroundMode.Shaded));
+        }
```
[the remark this answers is not in the document]
Comment (tomlm) [reply src/Tests/Consolonia.Core.Tests/PixelBackgroundTests.cs:24]: cool, didn't know you could do tha.t

**Comment by tomlm** [comment]
I don't know if you like the name of **Consolonia.NUnit** but it aligns with **Avalonia.Headless.Tests.NUnit/XUnit.** 
Alternate name could be **Consolonia.Tests.NUnit?**

**Comment by jinek** [comment]
> I don't know if you like the name of **Consolonia.NUnit** but it aligns with **Avalonia.Headless.Tests.NUnit/XUnit.** 
> Alternate name could be **Consolonia.Tests.NUnit?**  

Second can be found using keyword "test".
But second is mire aligned with avalonia, no? I'm fine with both names. May be leave it as is.

**Review by jinek — APPROVED** [review]

**Pull request merged**